pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -320,6 +320,17 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, r.Update(context.TODO(), cd)
 	}
 
+	// Set platform label on the ClusterDeployment
+	platform := getClusterPlatform(cd)
+	if cd.Labels == nil {
+		cd.Labels = make(map[string]string)
+	}
+	cdPlatform, cdPlatformLabelExists := cd.Labels[hivev1.HiveClusterPlatformLabel]
+	if !cdPlatformLabelExists || cdPlatform != platform {
```

Comment:
The check of `!cdPlatformLabelExists` is not necessary. If the label does not exist, then `cdPlatform` will be an empty string.